yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -289,6 +269,22 @@ func (p *ParserIssueDetector) getDDLIssues(query string) ([]QueryIssue, error) {
 			issues[i].SqlStatement = query
 		}
 	}
+
+	if _, ok := ddlObj.(*queryparser.Object); ok { // In case the DDL doesn't have any processor skip checking generic issues
```

Comment:
a ddlobject is an interface that already adheres to `GetObjectName` and `GetSchemaName`. So why do we need to additionally check if you can type cast it to a `queryparser.Object`. As long as ddlObject is of type `DDLObject interface`, you should be good, no?

Reply:
So there were reasons to add this check here-
1. To handle the scenario where the stmt coming to getDDLIssues is DML stmt which can be in case of PLPGQueries where we call internal getAllIssues(), so with that generic function is getting called for the select stmt twice giving the duplicate issues. - This can be solved by just unique issues by doing proper cmp.Equal()
2. To handle the scenario where the ObjectType is not known in this code path (the case where DDLObject is not implemented for some DDL types yet) so it might return. the objectType - `OBJECT` (NoOpProcessor). For such case, the analyze code to convertIssue toAnalyzeIssue() where we set invalidCount of objectname based on ObjectTYpe will fail with nil pointer as it doesn't know about it. - This can also be solved by handling it properly in that function.

Reply:
As discussed,  changedthis condition with isDDL in the starting of getDDLIssues()